scipy/signal/signaltools.py

```diff
@@ -463,6 +460,9 @@ def convolve(in1, in2, mode='full'):
     volume = asarray(in1)
     kernel = asarray(in2)
 
+    if not _is_valid_mode_shapes(mode, volume.shape, kernel.shape):
+        volume, kernel = kernel, volume
```

Comment:
Yes, it will fail the check regardless, but an `Exception` is no longer thrown.  That is the important point.